outlines/exploring-graphiql-the-in-browser-ide-for-graphql.md

```diff
@@ -0,0 +1,187 @@
+
+E. Documentation: GraphiQL generates documentation for your schema, making it easy to understand and explore.
+
+[Screenshot the documentation interface]
+
+## Testing out the Interface
+
+A. Writing the query
+
+B. Integrating Variables with Your Query
+
+C. Utilising fragments to aid query construction
+
+D. Executing/Testing the query: Sending requests and handling responses
+
+[take a screenshot and add code for each of the step above]
+
+## Debugging GraphQL Queries with GraphiQL
+
+### Practical approach to debugging
+
+A. Setting up your GraphQL endpoint in the GraphiQl (add a link to this section in the exploring GraphiQL article). Launch GraphiQl and configure the endpoint.
+
+(add screenshot and codebase for setting up the GraphQl endpoint)
+
+B. Write your query in the Query editor pane
+
+```graphql
+# your query
+```
+
+C. Check for Syntax errors: GraphiQL will automatically validate your query and highlight any syntax errors in red as you type
+Example: missing curly braces
+
+```graphql
+# code sample with this error
+```
+
+(Screenshot of the error interface)
+
+D. Inspect your Schema: Use the Documentation Explorer pane to understand your schema's available types, fields, and arguments. Click on the types and fields to view their details
+
+(screenshot each of this interface)
+
+E. Test with variables: Add variables in the Variables Editor to make your query dynamic.
+
+```graphql
+# query showing the variable added
+```
+
+(screenshot the variable pane)
+
+F. Analysing the response
+
+- Inspect the response in the Result Pane to ensure you're getting the expected data
+
+```json
+// sample of successful response
+```
+
+- Explain an instance where there is an error and how users can use the error messages to identify the issues.
+
+```json
+// sample of an error response
+```
+
+### Debugging Strategies
+
+A. Errors that can lead to bugs in the Queries
+
+- Syntax
+
+(Add a screenshot of a query in the editor with a syntax error)
+
+- Type Mismatches
+
+(Add a screenshot of a query in the editor with a type mismatch error)
+
+- Field Resolution issues.
+
+(Add a diagram analysing the field resolution issue)
+
+B. Error Messages
+
+- Understanding and interpreting GraphQL error messages
+
+C. Network Tab in Developer Tools
+
+- Analyzing network requests and responses for additional insights.
+
+### Best Practices
+
+A. Writing clear and well-structured queries.
+
+B. Using modular queries, i.e. breaking down complex queries into smaller and manageable chunks.
+
+C. Implement thorough/consistent query testing strategies to detect errors early.
+
+D. Adding in-line comments in your Query
+
+## Alternatives to GraphiQL
+
+A. Explain that there are also alternatives to GraphiQL. There are other GraphQL playgrounds like Insomnia, Altair client and so on
+
+B. Add a table to compare GraphiQl to Altair Client and Insomnia
+
+## Conclusion
+
+A. Summary of what we discussed
+
+B. Final thoughts on the subject
+
+## Resources
```

Comment:
Drop this section

Reply:
Alright, I will.

Reply:
> Drop this section

Including the conclusion?

Reply:
## Resources section

Reply:
> ## Resources section

Done

Reply:
its still there

Reply:
> its still there

I have resolved it in this new commit. Thank you!